src/Microsoft.NetCore.Analyzers/Core/Runtime/DoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectly.Fixer.cs

```diff
@@ -21,9 +30,77 @@ public sealed override FixAllProvider GetFixAllProvider()
 
         public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // Fixer not yet implemented.
+            // Only one diagnostic expected 
+            Diagnostic diagnostic = context.Diagnostics.Single();
+
+            // The fixer is only implemented for "Enumerable.First"
+            if (!diagnostic.Properties.TryGetValue("method", out var method) || method != "First")
+            {
+                return Task.CompletedTask;
+            }
+
+            string title = SystemRuntimeAnalyzersResources.UseIndexerInstead;
+
+            context.RegisterCodeFix(new MyCodeAction(title,
```

Comment:
I think we should offer a code fix only for cases where the compile time type of the invocation receiver/instance has an indexer property with single integer parameter. That way we cover majority of cases and guarantee that the code fix will not break user code.

Reply:
Can you add pragma disable for it in source instead of blanket disable all diagnostics in new code?

Reply:
As I said above, having the properties check ensures that the fixer was invoked by the correct analyzer. Namely the one that has done all the right checks, making sure that the type of the collection is correct as well as the fact that the `First` function has a single argument.

I know I've asked about this before and you said that it was necessary to duplicate the checks in the fixer. The difference now is that properties check.

So is that sufficient? Or do you still think the checks should be duplicated? If yes, I'll do it right away.

Reply:
I got confused into assuming that fix is offered for cases where the type did not have indexers, and hence led to compile time errors. I will resolve this conversation, as that does not seem to be the case.

Reply:
I got confused into assuming that fix is offered for cases where the type did not have indexers, and hence led to compile time errors. I will resolve this conversation, as that does not seem to be the case.